test/modules/TestResultCoordinator/Reports/EdgeHubRestartTest/EdgeHubRestartDirectMethodReport.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TestResultCoordinator.Reports.EdgeHubRestartTest
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    /// <summary>
+    /// This is a counting report to show test result counts, e.g. expect and match counts; and contains a list of unmatched test results.
+    /// </summary>
+    class EdgeHubRestartDirectMethodReport : TestResultReportBase
+    {
+        public EdgeHubRestartDirectMethodReport(
+            string trackingId,
+            string resultType,
+            bool isPassing,
+            long passedMessageCount,
+            Dictionary<string, ulong> messageCount,
+            Dictionary<HttpStatusCode, List<TimeSpan>> completedStatusHistogram,
```

Comment:
if i have 3 results - 2 results has status code "x" and both completed test in 5s and 1 result also has status code "x" and completed in 3s.  How completedStatusHistogram to support?  Again, what question do we want to answer by using histogram?  I don't want the report to grow too big.  If you want to share this information, please just write to TRC log.

Reply:
The `completedStatusHistrogram` is used to record of all the sender's result categorized by the status code as a key. In a passing case, we want to only look at the bucket of HTTP.OK and find the max of the timespan within that bucket. If the timespan is less than a minute the test passed.

The idea behind the histogram is that you can have a statistic result of how long the EdgeHub takes to restart.  For an example, imagine a case scenario, case 1: mean=50s, variance=0.1; I won't not have a confidence that the edgeHub "restarts" under 1 minutes since the variance is so low with mean pushing the boundary of our limit ( ideally this case should fail the test ).

With that said, I can have the report not print out all the timespan that it records, but instead a statistic of each bucket, namely {min, max, mean, mid, variance}

Reply:
Do we get this information from metric?

we use hard-limit used in test to timeout invoking DM call or send message call and decide if that test run is failed; not based on mean and variation.  Please simplify this report if there is no strong reason to have it.

Reply:
Please remove "completedStatusHistogram" here; you can write this data to log in report generator.